specs/host_list_view.adoc

```diff
@@ -0,0 +1,135 @@
+// vim: tw=79
+
+= Tendrl Definitions: Display list of hosts discovered by Tendrl
+
+Display list of hosts discovered by Tendrl, whether they currently associated with a cluster or not. The list should have functionality for 
+
+* filtering by name, cluster, role and status.
+* sorting by name.
+
+Host list will be sorted by name by default.
+
+== Problem description
+
+Tendrl should show list of hosts discovered by Tendrl along with their details.
+
+The details shown for each host in UI is -
+
+* Host Name
+* Status
+* IP Address
+* Storage Used
+* CPU Utilisation
+* Memory Used
+* Cluster Name
+* Role
+* Alerts
+
+API supports present now is for -
+
+* fqdn(Host Name)
+
+== Use Cases
+
+Users can see the list of host present in Tendrl. They can also 
+
+* filter by name, cluster, role and status.
+* sort by name.
+
+== Proposed change
+
+* Create a view for listing host details.
+* For showing used storage, cpu utilisation and memory used, use donut and trending chart component.
+* This view will use generate-list component for listing host. The list data will be passed to this component from the controller.
+* Implement filer and sorting functionality.
+* Filter parameter(name/cluster/role/status) should be sent to API to fetch the filtered list of host.
+* Sorting parameter(name and asc/desc) should be sent to API to fetch sorted list of host.
+
+=== Alternatives
+
+Host list can have its own html template instead of using generate-list component. But the code will be redundant as the same functionality and layout is going to be used in all list views.
```

Comment:
How do we achieve clickability of certain column items say hosts and pools under cluster lists should link to hosts and pool lists respectively. Is this flexibility achievable using this mechanism?